include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -568,6 +581,24 @@ struct erfInv_helper<FloatingPoint NBL_PARTIAL_REQ_BOT(concepts::FloatingPointSc
 // 	}
 // };
 
+// log2-gamma function
+template<typename T>
+NBL_PARTIAL_REQ_TOP(concepts::FloatingPointScalar<T>)
+struct l2gamma_helper<T NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
+{
+	// modified Stirling's approximation for log2 up to 3rd polynomial
+	static T __call(T x)
+	{
+		const T thresholds[3] = {5e4, 1e19, 1e19};
+		if (x > thresholds[findLSB(uint32_t(sizeof(T)))])
+			return bit_cast<T>(numeric_limits<T>::infinity);
+
+		const T l2x = log2_helper<T>::__call(x);
+		const T one_over_ln2 = T(1.44269504088896);
+		return (x - T(0.5)) * l2x - one_over_ln2 * x + T(1.32574806473616) + one_over_ln2 / (T(12.0) * x) + one_over_ln2 / (T(360.0) * x * x * x);
```

Comment:
use the "efficient way to write a polynomial, so
```cpp
const T r = T(1)/x;
const T r2 = r*r;
return (x-T(0.5))*l2x + T(1.32574806473616) + one_over_ln2*((T(1.0/360.0)*r2+T(1.0/12.0))*r-x);
```

where you add polynomial reciprocal terms on the inside of the bracket

Btw for float32 please add more terms as per the IEOS sequences I linked you